Jim-Hodapp-Coaching/refactor-platform-fe
# Create coaching sessions page

## Description

This PR implements a baseline coaching sessions page to build further upon once the resource APIs in entity_api + controllers from the backend get implemented. The specifics of each UI section will be much more finely implemented at this time in line with the desired feature set for the first MVP.

This is not meant to be a polished page yet, just a sane-enough baseline to set a UI vision and build upon it.

#### GitHub Issue: Closes #6 

### Changes
* Implements a baseline coaching sessions page where a coach and coachee will spend most of their time for this platform
* Introduces several React UI components from [@shadcn](https://ui.shadcn.com/docs/components/)

### UI Screenshot

<img width="1780" alt="Screenshot 2024-03-09 at 00 40 15" src="https://github.com/Jim-Hodapp-Coaching/refactor-platform-fe/assets/3219120/16e9df34-4f87-492e-96d9-a1e4fb2bb88d">

### Testing Strategy
1. Start the frontend `npm run dev`
2. Navigate your web browser to http://localhost:3000/coaching-sessions
3. Observe the coaching-sessions page renders with no React hydration errors


### Concerns
None
